internal/cli/cmd/cluster/artifact.go

```diff
@@ -102,24 +125,160 @@ func newArtifactDownloadCmd() *cobra.Command {
 		}
 		defer cli.Close()
 
-		reader, err := storage.ResolveArtifactStream(ctx, cli, namespace, src)
-		if err != nil {
+		if err := writeArtifact(ctx, cli, namespace, src, dest); err != nil {
 			return err
 		}
-		defer reader.Close()
 
-		w, err := os.Create(dest)
+		fmt.Fprintf(console.Stdout(ctx), "Downloaded %s (namespace %s) to %s\n", src, namespace, dest)
+
+		return nil
+	})
+}
+
+func newArtifactCacheURLCmd() *cobra.Command {
+	var renew bool
+
+	return fncobra.Cmd(&cobra.Command{
+		Use:   "cache-url [target url] [destination filename] { --renew }",
+		Short: "Download an arbitrary URL using a pull-through cache.",
+		Long: `Download the content from an arbitrary URL and cache it for fast access.
+
+If the content is already present in the artifact cache for the given URL it will be used instead.
+
+The content at the URL is assumed to be immutable.`,
+		Args: cobra.ExactArgs(2),
+	}).WithFlags(func(flags *pflag.FlagSet) {
+		flags.BoolVar(&renew, "renew", false, "Force-download from the source and update the cached content.")
```

Comment:
hmm, should we use a staleness limit instead (e.g. --max-age)?

Reply:
Good idea. Introduced `--max_age` instead of renew.